### PR TITLE
[DCOS-2874] Fixed a bug where `dcos task log` wasn't working with core.mesos_mast…

### PR DESCRIPTION
**Pull request by mgummelt**
…er_url

### CODE REVIEW

File: dcos/mesos.py [hunk 1]
```diff
@@ -67,20 +67,26 @@ def master_url(self, path):
...
-    def slave_url(self, slave_id, path):
-        """ Create a slave URL
+    def slave_url(self, slave_id, private_url, path):
+        """Create a slave URL
 
         :param slave_id: slave ID
         :type slave_id: str
+        :param private_url: The slave's private URL derived from its
+          pid.  Used when we're accessing mesos directly, rather than
+          through DCOS.
```
Comment (tamarrow) [inline dcos/mesos.py]: Can we fix the spacing to be aligned after the `:` for consistency with file?

File: dcos/mesos.py [hunk 2]
```diff
@@ -92,16 +98,20 @@ def get_master_state(self):
...
 
-    def get_slave_state(self, slave_id):
+    def get_slave_state(self, slave_id, private_url):
         """Get the Mesos slave state json object
 
         :param slave_id: slave ID
         :type slave_id: str
+        :param private_url: The slave's private URL derived from its
+          pid.  Used when we're accessing mesos directly, rather than
+          through DCOS.
```
Comment (tamarrow) [inline dcos/mesos.py]: re spacing problem here too

File: dcos/mesos.py [hunk 3]
```diff
@@ -114,11 +124,17 @@ def get_state_summary(self):
...
+    def slave_file_read(self, slave_id, private_url, path, offset, length):
         """ See the master_file_read() docs
 
+        :param slave_id: slave ID
+        :type slave_id: str
         :param path: absolute path to read
         :type path: str
+        :param private_url: The slave's private URL derived from its
+          pid.  Used when we're accessing mesos directly, rather than
+          through DCOS.
```
Comment (tamarrow) [inline dcos/mesos.py]: and here

**Comment by mgummelt** [comment]
fixed spacing

**Comment by tamarrow** [comment]
@mgummelt Can you shorten the commit message to make it nicer for the release text? otherwise LGTM. merge when tests pass.

**Comment by mgummelt** [comment]
Do you have a suggestion?

**Comment by tamarrow** [comment]
something similar to the jira title? `[DCOS-2874] add raw mesos support for `dcos task log``

**Comment by mgummelt** [comment]
It's unclear what "raw mesos support" means.  Why is ~70 characters too long?

**Comment by mgummelt** [comment]
Not worth arguing over.  I'll change it.

**Pull request merged**